zipline/algorithm.py

```diff
@@ -603,14 +603,26 @@ def run(self, data=None, overwrite_sim_params=True):
                 data = data.swapaxes(0, 2)
 
             if isinstance(data, pd.Panel):
+                # Guard against tz-naive index.
+                if data.major_axis.tz is None:
+                    data.major_axis = data.major_axis.tz_localize('UTC')
+
                 # For compatibility with existing examples allow start/end
-                # to be inferred.
+                # to be inferred. Start and end have to be midnight UTC.
                 if overwrite_sim_params:
                     self.sim_params = self.sim_params.create_new(
-                        data.major_axis[0],
-                        data.major_axis[-1]
+                        normalize_date(data.major_axis[0].tz_convert('UTC')),
```

Comment:
Should we be normalizing to midnight nowadays, or finding the session for the particular minute according to the trading calendar?